testing/test_parser.py

```diff
@@ -0,0 +1,156 @@
+import ast
+from itertools import zip_longest
+from pathlib import Path
+from typing import List, Tuple
+
+import pytest
+import pytest_check as check
+from flake8_annotations import Argument, Function, FunctionVisitor
+from testing import parser_object_attributes
+
+
+class TestArgumentParsing:
+    """Test for proper argument parsing from source."""
+
+    src_filepath = Path("./testing/code/all_args.py")
+    with src_filepath.open("r", encoding="utf-8") as f:
+        src = f.read()
+
+    tree = ast.parse(src)
+    lines = src.splitlines()
+    visitor = FunctionVisitor(lines)
+    visitor.visit(tree)
+
+    @pytest.fixture(params=parser_object_attributes.parsed_arguments.keys())
+    def argument_lists(self, request) -> Tuple[List[Argument], List[Argument], str]:  # noqa
+        """
+        Build a pair of lists of arguments to compare and return as a (truth, parsed) tuple.
+
+        `parser_object_attributes.parsed_arguments` is a dictionary of the arguments that should be
+        parsed out of the testing source code:
+          * Keys are the function name, as str
+          * Values are a list of Argument objects that should be parsed from the function definition
+
+        A list of parsed Argument objects is taken from the class-level source parser
+
+        The function name is also returned in order to provide a more verbose message for a failed
+        assertion
+
+        Note: For testing purposes, Argument lineno and col_offset are ignored so these are set to
+        dummy values in the truth dictionary
+        """
+        truth_arguments = parser_object_attributes.parsed_arguments[request.param]
+
+        # Find the function in our source parser & get the list of Argument objects
+        for function in self.visitor.function_definitions:
+            if function.name == request.param:
+                parsed_arguments = function.args
+                break
+        else:
+            # We shouldn't ever get here, but add a catch-all just in case
+            parsed_arguments = None
+
+        return truth_arguments, parsed_arguments, request.param
+
+    def test_argument_parsing(
+        self, argument_lists: Tuple[List[Argument], List[Argument], str]
+    ) -> None:
+        """
+        Test argument parsing of the testing source code.
+
+        Argument objects are provided as a tuple of (truth, source) lists
+        """
+        for truth_arg, parsed_arg in zip_longest(argument_lists[0], argument_lists[1]):
```

Comment:
Just an idea. I don't know if it's necessarily clearer but it is shorter.
```suggestion
        for truth_arg, parsed_arg in zip_longest(*argument_lists[:2]):
```